src/face_op.cpp

```diff
@@ -192,38 +246,104 @@ void Manifold::Impl::Face2Tri(const Vec<int>& faceEdge,
   CreateHalfedges(triVerts);
 }
 
-/**
- * Returns a set of 2D polygons formed by the input projection of the vertices
- * of the list of Halfedges, which must be an even-manifold, meaning each vert
- * must be referenced the same number of times as a startVert and endVert.
- */
-PolygonsIdx Manifold::Impl::Face2Polygons(VecView<Halfedge>::IterC start,
-                                          VecView<Halfedge>::IterC end,
-                                          mat2x3 projection) const {
-  std::multimap<int, int> vert_edge;
-  for (auto edge = start; edge != end; ++edge) {
-    vert_edge.emplace(
-        std::make_pair(edge->startVert, static_cast<int>(edge - start)));
-  }
+void Manifold::Impl::FlattenFaces() {
+  Vec<size_t> edgeFace(halfedge_.size());
+  const size_t remove = std::numeric_limits<size_t>::max();
+  const size_t numTri = NumTri();
+  const auto policy = autoPolicy(numTri);
 
-  PolygonsIdx polys;
-  int startEdge = 0;
-  int thisEdge = startEdge;
-  while (1) {
-    if (thisEdge == startEdge) {
-      if (vert_edge.empty()) break;
-      startEdge = vert_edge.begin()->second;
-      thisEdge = startEdge;
-      polys.push_back({});
+  std::vector<std::atomic<int>> vertDegree(NumVert());
+  for_each(policy, vertDegree.begin(), vertDegree.end(),
+           [](auto& v) { v.store(0); });
+
+  for_each_n(policy, countAt(0_uz), numTri,
+             [&edgeFace, &vertDegree, this](size_t tri) {
+               for (const int i : {0, 1, 2}) {
+                 const int edge = 3 * tri + i;
+                 const int pair = halfedge_[edge].pairedHalfedge;
+                 const auto& ref = meshRelation_.triRef[tri];
+                 if (ref.SameFace(meshRelation_.triRef[pair / 3])) {
+                   edgeFace[edge] = remove;
+                 } else {
+                   edgeFace[edge] = (static_cast<size_t>(ref.meshID) << 32) +
```

Comment:
For wasm, `size_t` is 32 bit. We should probably explicitly make this `uint64_t`.